services/spar/src/Spar/API.hs

```diff
@@ -173,9 +173,15 @@ idpDelete :: Maybe UserId -> SAML.IdPId -> Spar NoContent
 idpDelete zusr idpid = withDebugLog "idpDelete" (const Nothing) $ do
     idp <- SAML.getIdPConfig idpid
     authorizeIdP zusr idp
+
+    let issuer = idp ^. SAML.idpMetadata . SAML.edIssuer
+        team = idp ^. SAML.idpExtraInfo
+
+    -- fail if idp is not empty
+    idpIsEmpty <- wrapMonadClient $ isNothing <$> Data.getSAMLAnyUserByIssuer issuer
```

Comment:
Why is there a map here, when we are getting a single user?

Reply:
> Why is there a map here, when we are getting a single user?

The return value of `Data.getSAMLAnyUserByIssuer` is `m (Maybe UserId)` for some functor `m`.  `<$>` maps into any functor, so in this case it's not `[]`, but `m`.  Inside the `m`, we have a maybe value, which is transformed into a boolean by [`isNothing`](https://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Maybe.html#v:isNothing).

Is that a good answer?  And am I answering the right question?

Reply:
yep. works for me.